hcloud/resource_hcloud_reversedns.go

```
@@ -0,0 +1,226 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+func resourceReverseDNS() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDNSRead,
+		Create: resourceReverseDNSCreate,
+		Delete: resourceReverseDNSDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns_ptr": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceReverseDNSRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ipAddress := string(d.Get("ip_address").(string))
+	server, floatingIP, err := lookupRDNSID(d.Id(), client)
+
+	switch {
+	case err != nil:
+		log.Printf("[WARN] Invalid id (%s), removing from state: %s", d.Id(), err)
+		d.SetId("")
+		return nil
+	}
+	ip := net.ParseIP(d.Get("ip_address").(string))
+	if ip == nil {
+		log.Printf("[ERR] The ip you provide (%s) is not valid", ipAddress)
+		return nil
```

return the error instead of logging it, to inform the user.